paymentsheet/src/main/java/com/stripe/android/paymentelement/confirmation/intent/IntentConfirmationInterceptor.kt

```diff
@@ -302,31 +330,50 @@ internal class DefaultIntentConfirmationInterceptor @Inject constructor(
             productUsage = productUsage,
         )
 
-        return createPaymentMethod(params).fold(
-            onSuccess = { paymentMethod ->
-                handleDeferred(
-                    intentConfiguration = intentConfiguration,
-                    paymentMethod = paymentMethod,
-                    paymentMethodOptionsParams = paymentMethodOptionsParams,
-                    paymentMethodExtraParams = paymentMethodExtraParams,
-                    shippingValues = shippingValues,
-                    shouldSavePaymentMethod = customerRequestedSave || shouldSavePaymentMethod(
-                        paymentMethodOptionsParams = paymentMethodOptionsParams,
-                        intentConfiguration = intentConfiguration
-                    ),
-                    hCaptchaToken = null
-                )
-            },
-            onFailure = { error ->
-                NextStep.Fail(
-                    cause = error,
-                    message = error.stripeErrorMessage(),
-                )
+        val deferredIntentCallback: DeferredIntentCallback? =
+            if (intentConfiguration.intentBehavior is PaymentSheet.IntentConfiguration.IntentBehavior.Default) {
+                waitForIntentCallback()
+            } else {
+                null
             }
-        )
+        return if (deferredIntentCallback is DeferredIntentCallback.CreateIntentWithConfirmationToken) {
```

Comment:
It's probably time for an abstraction here. I think this would be easier to test if we had a thing that did this work, and we could test each of those in isolation.

I think we should work on that as pre work to the functionality this PR introduces.

Let's chat about next steps for this offline!